ci-services/appveyor.js

```diff
@@ -0,0 +1,25 @@
+'use strict'
+
+const env = process.env
+
+/**
+ * Should update the `package-lock.json`
+ * Thanks @selbyk
+ */
+function shouldUpdate () {
+  let re = /^(chore|fix)\(package\): update [^ ]+ to version.*$/mi
+  return re.test(env.APPVEYOR_REPO_COMMIT_MESSAGE)
+}
+
+module.exports = {
+  repoSlug: env.APPVEYOR_REPO_NAME,
+  branchName: env.APPVEYOR_REPO_BRANCH,
+  firstPush: shouldUpdate(),
+  correctBuild: !env.APPVEYOR_PULL_REQUEST_NUMBER,
+  uploadBuild: true,
+  // Used to ignore output when staging the updated lockfile
+  // The Windows equivalent of '2>/dev/null || true' is '2>NUL || (exit 0)'
+  ignoreOutput: '2>NUL || (exit 0)',
+  // Multi-line update message because committing with linebreaks does not work.
+  updateMessage: '-m "chore(package): update lockfile" -m "https://npm.im/greenkeeper-lockfile"'
```

Comment:
Changing updateMessage is necessary. New line characters \n\n don't work in AppVeyor. Here is a failed build when this option is not set: https://ci.appveyor.com/project/patkub/test/build/1.0.5
https://github.com/patkub/greenkeeper-lockfile/tree/appveyor-ci-service-no-update-message